mantidproject/mantid
# Crash reporting service

This adds crash reporting to the last chance error handler.
**To test:**

- To test you first need to provoke an uncaught exception. One method is here https://github.com/mantidproject/mantid/issues/18990

- When you open mantid turn usage reporting on in the splashscreen. This is no longer necessary and error reporting should work if usage reporting is turned off. **However where the usage reporting gets its url from has been changed so this should be checked to still send usage reports to the right places.**

- Check that the usage reporting startup message is sent correctly it should go to errorreports.mantidproject.org/api/error

- Confirm that when you cause an uncaught exception a dialog window appears asking you whether you wish to continue using mantid and what information you want to share.

- Check that Mantid continues or terminates correctly.

- Check that the correct information (including importantly nothing if you decide to share nothing) is sent in a post request. 

- Also requires a code review.



ToDo
Fixes #21709 . 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
